platform/posix/clock_posix.c

```diff
@@ -37,24 +37,28 @@
 /*
  * Time conversion constants.
  */
-#define NANOSECONDS_PER_MILLISECOND    ( 1000000 )    /**< @brief Nanoseconds per millisecond. */
-#define MILLISECONDS_PER_SECOND        ( 1000 )       /**< @brief Milliseconds per second. */
+#define NANOSECONDS_PER_MILLISECOND    ( 1000000L )    /**< @brief Nanoseconds per millisecond. */
+#define MILLISECONDS_PER_SECOND        ( 1000L )       /**< @brief Milliseconds per second. */
 
 /*-----------------------------------------------------------*/
 
 uint32_t Clock_GetTimeMs( void )
 {
-    uint32_t timeMs;
+    int64_t timeMs;
     struct timespec timeSpec;
 
     /* Get the MONOTONIC time. */
-    clock_gettime( CLOCK_MONOTONIC, &timeSpec );
+    ( void ) clock_gettime( CLOCK_MONOTONIC, &timeSpec );
 
     /* Calculate the milliseconds from timespec. */
-    timeMs = ( uint32_t ) ( timeSpec.tv_sec * 1000 )
-             + ( uint32_t ) ( timeSpec.tv_nsec / ( 1000 * 1000 ) );
+    timeMs = ( timeSpec.tv_sec * MILLISECONDS_PER_SECOND )
+             + ( timeSpec.tv_nsec / NANOSECONDS_PER_MILLISECOND );
 
-    return timeMs;
+    /* Libraries need only the lower 32 bits of the time in milliseconds, since
+     * this function is used only for a calculating the time difference.
+     * Also, the possible overflows of this time value is handled in the
+     * libraries. */
+    return ( uint32_t ) timeMs;
```

Comment:
Does this cast not create a MISRA violation?

Reply:
No. Misra doesn't complain about this as it is an explicit cast.